packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -113,10 +115,17 @@ class Observatory {
     });
   }
 
-  Future<Response> isolateReload(String isolateId) {
-    return sendRequest('isolateReload', <String, dynamic>{
-      'isolateId': isolateId
-    });
+  Future<bool> reloadSources(String isolateId) async {
+    Future<Event> future = onIsolateEvent
+      .where((Event event) => event.kind == 'IsolateReload')
+      .single;
+
+    await sendRequest('_reloadSources', <String, dynamic>{ 'isolateId': isolateId });
+    Event event = await future.timeout(new Duration(seconds: 20));
```

Comment:
Because `future` is backed by a stream, I think when a timeout occurs it will leak the subscription to `onIsolateEvent`. You may need a `try/finally` and cancel the subscription explicitly in the `finally` block.